Automattic/wp-calypso
# Unify Design Preview: Display both Colors & Fonts with Navigator

Related to https://github.com/Automattic/wp-calypso/issues/78419

## Proposed Changes

* Move `navigator-buttons`, `navigator-header`, and `navigator-item-group` into `@automattic/onboarding` package
* Implement `NavigatorScreens`, `useNavigatorButtons`, and `useNavigatorScreens` to display navigator buttons, and navigate between screens.
* Hide the back button on the top-left corner (next to the WP logo) when you're in either Colors or Fonts screen

### Screenshots

**Style Variations**

| Desktop | Mobile |
| - | - |
| ![image](https://github.com/Automattic/wp-calypso/assets/13596067/6a8fb6d7-87af-4105-8705-d0bf1f440a4a) | ![image](https://github.com/Automattic/wp-calypso/assets/13596067/6b330e95-ff03-44c0-9e49-7010042da87c) |

**Colors**
| Desktop | Mobile |
| - | - |
| ![image](https://github.com/Automattic/wp-calypso/assets/13596067/75ad638c-0fcf-4788-9111-018d277d5929) | ![image](https://github.com/Automattic/wp-calypso/assets/13596067/f54350af-edad-4a5c-8607-84d4eda122e4) |

**Fonts**
| Desktop | Mobile |
| - | - |
| ![image](https://github.com/Automattic/wp-calypso/assets/13596067/d5a5ec5e-6e18-4020-b988-feaca4bff8ed) | ![image](https://github.com/Automattic/wp-calypso/assets/13596067/10b60266-b1a0-47c8-a77b-bac4370409cc) |

**Both**

| Main Screen | Colors Screen | Fonts Screen |
| - | - | - |
| ![image](https://github.com/Automattic/wp-calypso/assets/13596067/7cc71c0b-69d7-4e8d-8286-08fa73d28e09) | ![image](https://github.com/Automattic/wp-calypso/assets/13596067/70e91d50-10b2-4cc4-98b0-393530e9a60d) | ![image](https://github.com/Automattic/wp-calypso/assets/13596067/fa692d4c-b1e7-4698-9ba4-f3019672965f) |

The mobile layout for both Colors & Fonts will be in the follow-up PR

## Testing Instructions



* Go to URLs below
  * `/setup?siteSlug=<your_site>`: Fonts only
  * `/setup?siteSlug=<your_site>&flags=signup/design-picker-preview-colors`: Colors & Fonts
* Continue until you land on the Design Picker
* Preview the design with style variation, everything should be good as before
* Preview the design without the style variation
* On the preview screen,
  * You're able to see both Colors & Fonts
  * Select Colors to navigate to Colors screen
  * Select Fonts to navigate to Fonts screen
  * If you're in either Colors & Fonts screen, the back button on the top-left corner (next to the WP logo) will be hidden

## Pre-merge Checklist
